packages/flutter/lib/src/widgets/snapshot_widget.dart

```diff
@@ -356,19 +365,21 @@ abstract class SnapshotPainter extends ChangeNotifier  {
   /// [SnapshotPainter] paints the snapshot. This must account for the fact that the image
   /// width and height will be given in physical pixels, while the image must be painted with
   /// device independent pixels. That is, the width and height of the image is the widget and
-  /// height of the provided `size`, multiplied by the `pixelRatio`:
+  /// height of the provided `size`, multiplied by the `pixelRatio`. In addition, the actual
+  /// size of the scene captured by the `image` is not `image.width` or `image.height`, but
+  /// indeed `srcSize`, because the former is a rounded inaccurate integer:
   ///
   /// ```dart
-  /// void paint(PaintingContext context, Offset offset, Size size, ui.Image image, double pixelRatio) {
-  ///   final Rect src = Rect.fromLTWH(0, 0, image.width.toDouble(), image.height.toDouble());
+  /// void paint(PaintingContext context, Offset offset, Size size, ui.Image image, Size srcSize, double pixelRatio) {
+  ///   final Rect src = Rect.fromLTWH(0, 0, srcSize.width, srcSize.height);
   ///   final Rect dst = Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height);
   ///   final Paint paint = Paint()
   ///     ..filterQuality = FilterQuality.low;
   ///   context.canvas.drawImageRect(image, src, dst, paint);
   /// }
   /// ```
   /// {@end-tool}
-  void paintSnapshot(PaintingContext context, Offset offset, Size size, ui.Image image, double pixelRatio);
+  void paintSnapshot(PaintingContext context, Offset offset, Size size, ui.Image image, Size srcSize, double pixelRatio);
```

Comment:
srcSize: In accordance with our style guide, please avoid abbreviations, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-abbreviations

Reply:
Then sourceSize?